src/renderer/src/hooks/useBlockchainActions.ts

```diff
@@ -0,0 +1,156 @@
+import { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+import { TBlockchainServiceKey } from '@renderer/@types/blockchain'
+import { TWalletType } from '@renderer/@types/store'
+import { UtilsHelper } from '@renderer/helpers/UtilsHelper'
+import { accountReducerActions } from '@renderer/store/account/AccountReducer'
+import { selectBsAggregator } from '@renderer/store/blockchain/SelectorBlockchain'
+import { walletReducerActions } from '@renderer/store/wallet/WalletReducer'
+import * as uuid from 'uuid'
+
+import { Account } from '../store/account/Account'
+import { selectAccounts } from '../store/account/SelectorAccount'
+import { Wallet } from '../store/wallet/Wallet'
+
+import useSelectorRef from './useSelectorRef'
+
+export type TAccountToImport = {
+  address: string
+  blockchain: TBlockchainServiceKey
+  wallet: Wallet
+  type: TWalletType
+  key?: string
+  name?: string
+}
+
+export type TAccountToCreate = {
+  wallet: Wallet
+  name: string
+  blockchain: TBlockchainServiceKey
+  index?: number
+}
+
+export type TWalletToCreate = {
+  name: string
+  walletType: TWalletType
+  mnemonic?: string
+}
+
+const colors = {
+  0: '#00ddb4',
+  1: '#22b1ff',
+  2: '#7c4bfe',
+  3: '#d355e7',
+  4: '#fe872f',
+  5: '#fedd5b',
+  6: '#91abbc',
+}
+
+export function useBlockchainActions() {
+  const dispatch = useDispatch()
+  const accounts = useSelectorRef(selectAccounts)
+  const bsAggregator = useSelectorRef(selectBsAggregator)
+  const encryptedPassword = useSelectorRef(state => state.settings.encryptedPassword)
+
+  const createWallet = useCallback(
+    async ({ name, walletType, mnemonic }: TWalletToCreate) => {
+      let encryptedMnemonic: string | undefined
+
+      if (walletType === 'standard') {
+        if (!mnemonic) throw new Error('Standard Wallet needs to have a security phrase')
+
+        encryptedMnemonic = await window.api.encryptBasedEncryptedSecret(mnemonic, encryptedPassword.current)
+      }
+
+      const id = uuid.v4()
+
+      const newWallet = new Wallet({
+        name,
+        walletType,
+        id,
+        encryptedMnemonic,
+      })
+
+      dispatch(walletReducerActions.saveWallet(newWallet.deserialize()))
+
+      return newWallet
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why this is happening? Let's discuss how to avoid this comments, maybe we can change the lint rules